spec/services/moderator/banish_user_spec.rb

```diff
@@ -0,0 +1,92 @@
+require "rails_helper"
+
+RSpec.describe Moderator::BanishUser, type: :service, vcr: {} do
+  let(:user) { create(:user, :with_complete_profile) }
+  let(:user_without_email) { create(:user, email: nil) }
+  let(:admin) { create(:user, :super_admin) }
+
+  describe "banish_user" do
+    it "unsubscribe user from newsletters if user has email address" do
+      allow(user).to receive(:unsubscribe_from_newsletters)
+      described_class.call_banish(user: user, admin: admin)
+      expect(user).to have_received(:unsubscribe_from_newsletters).once
+    end
+
+    context "with user profile info" do # Split into 2 to make rubocop happy
+      it "removes user profile details" do
+        expect do
+          described_class.call_banish(user: user, admin: admin)
+        end.to change(user, :twitter_username).to(nil).and change(user, :github_username).to(nil).
+          and change(user, :summary).to("").and change(user, :location).to("").
+          and change(user, :education).to("").and change(user, :employer_name).to("").
+          and change(user, :employer_url).to("").and change(user, :employment_title).to("").
+          and change(user, :mostly_work_with).to("").and change(user, :currently_learning).to("").
+          and change(user, :currently_hacking_on).to("").and change(user, :available_for).to("").
+          and change(user, :email_public).to(false)
+      end
+
+      it "removes a users social-media details" do
+        expect do
+          described_class.call_banish(user: user, admin: admin)
+        end.to change(user, :facebook_url).to(nil).and change(user, :dribbble_url).to(nil).
+          and change(user, :medium_url).to(nil).and change(user, :stackoverflow_url).to(nil).
+          and change(user, :behance_url).to(nil).and change(user, :linkedin_url).to(nil).
+          and change(user, :gitlab_url).to(nil).and change(user, :instagram_url).to(nil).
+          and change(user, :mastodon_url).to(nil).and change(user, :twitch_url).to(nil).
+          and change(user, :feed_url).to(nil).
+          and change { user.reload.profile_image_identifier }.to("https://thepracticaldev.s3.amazonaws.com/i/99mvlsfu5tfj9m7ku25d.png")
```

Comment:
where does this URL string come from?

Reply:
@rhymes Thanks for the quick review.
The URL string is hardcoded in [app/services/moderator/banish_user.rb:48](https://github.com/thepracticaldev/dev.to/blob/master/app/services/moderator/banish_user.rb#L48)

Reply:
sorry I didn't catch that, thank you!